x-pack/plugins/fleet/common/services/check_fleet_server_versions.ts

```diff
@@ -67,6 +67,10 @@ export const isAgentVersionLessThanFleetServer = (
   fleetServerAgents: Agent[],
   force = false
 ) => {
+  // For serverless it should not have any fleet server agents
+  if (fleetServerAgents.length === 0) {
```

Comment:
I am wondering if that condition is good enough or if we should have an explicit flag that we are in serverless.

Reply:
This seems fine to me. There's no way to have no fleet server agents and initiate an upgrade for an agent in cloud or self-hosted today so this block is only relevant to serverless projects.